compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.h

```diff
@@ -49,6 +50,22 @@ FailureOr<Flow::WorkloadBuilder> getWorkloadBuilder(OpBuilder &builder,
 LogicalResult simplifyDimOps(RewriterBase &rewriter,
                              const SmallVector<tensor::DimOp> &dimOps);
 
+/// Returns true if this is a fusable use, while fusing a root with its
+/// consumer.
+bool isFusableWithConsumer(OpOperand &fusedOperand,
+                           const llvm::SmallBitVector &rootOuterParallelLoops,
+                           FormDispatchRegionsPassOptions const &options);
+
+/// Method to check if the consumer of a use can be fused with its producer.
+bool isFusableWithProducer(
+    OpOperand &operand, const llvm::SmallBitVector &rootOuterParallelLoops,
+    FormDispatchRegionsPassOptions const &options,
+    std::optional<OpResult> overrideProducerResult = std::nullopt);
```

Comment:
When the producer is in a dispatch region, the defining op of the `OpOperand` will be the dispatch region.  `overrideProducerResult` is a way to circumvent this by acting as a replacement for the `OpOperand` defining op's result (it will get the matching yielded value inside the dispatch).

Reply:
Sorry, not able to follow. Can you give an example.

Reply:
In a case like:
```
%r = dispatch_region {
  %producer = op()
  flow.return %producer
}
%encoding = set_encoding %r
```
We want to check if `set_encoding` is fusable with `op`, but the OpOperand of `set_encoding` is `%r`. In this case, `overrideProducerResult` will be passed as `%producer`, and then the function will treat it as if it looks like:
```
%producer = op()
%encoding = set_encoding %producer
```
And decide whether the 2 ops are fusable.

Reply:
I can add this example to the comments on the function

Reply:
I removed the dependency on FormDispatchRegions, and just checked that producers are linalg ops, as we discussed offline.